lib/puppet/pops/loader/dependency_loader.rb

```diff
@@ -26,7 +26,7 @@ def initialize(parent_loader, name, dependency_loaders)
   def discover(type, error_collector = nil, name_authority = Puppet::Pops::Pcore::RUNTIME_NAME_AUTHORITY, &block)
     result = []
     @dependency_loaders.each { |loader| result.concat(loader.discover(type, error_collector, name_authority, &block)) }
-    result.concat(super)
+    result.concat(super).uniq!
```

Comment:
I'm not sure how to test this...it's a lot of setup, where you need to have multiple modules with the same parent module loader.